Device.md

```diff
@@ -0,0 +1,471 @@
+|R\_HW\_VERSION\_L|-|Yes|U8|002|a)|Minor Hardware version|Yes|
+|R\_ASSEMBLY\_VERSION|-|Yes|U8|003|a)|Version of the assembled components|Optional|
+|R\_CORE\_VERSION\_H|-|Yes|U8|004|a)|Major core version|Optional|
+|R\_CORE\_VERSION\_L|-|Yes|U8|005|a)|Minor core version|Optional|
+|R\_FW\_VERSION\_H|-|Yes|U8|006|a)|Major Firmware version of the application|Yes|
+|R\_FW\_VERSION\_L|-|Yes|U8|007|a)|Minor Firmware version of the application|Yes|
+|R\_TIMESTAMP\_SECOND|Yes|No|U32|008|0|System timestamp: seconds|Yes|
+|R\_TIMESTAMP\_MICRO|Yes|Yes|U16|009|0|System timestamp: microseconds|Optional|
+|R\_OPERATION\_CTRL|No|No|U8|010|b)|Configuration of the operation mode|c)|
+|R\_RESET\_DEV|No|No|U8|011|b)|Reset device and save non-volatile registers|Optional|
+|R\_DEVICE\_NAME|No|No|U8|012|b)|Name of the device given by the user|Optional|
+|R\_SERIAL\_NUMBER|No|No|U16|013|b)|Unique serial number of the device|Optional|
+|R\_CLOCK\_CONFIG|No|No|U8|014|b)|Synchronization clock configuration|Optional|
+|R\_TIMESTAMP\_OFFSET|No|No|U8|015|b)|Adds an offset if user updates the Timestamp|Optional|
+
+||a) These values are stored during factory process and are persistent, i.e., they cannot be changed by the user.<br>b) Check register notes on the specific register explanation<br>c) Only parts of the functionality is mandatory. Check register notes on the explanation.|
+| :- | :- |
+
+
+<!---
+Tables can be generated here https://www.tablesgenerator.com/html_tables
+Mermaid plots can be generated here: https://mermaid.live/
+--->
+
+#### **`R_WHO_AM_I` (U16) – Who Am I**
+
+Address: `000`
+
+Used to verify the identity of the device. A list of devices can be found at <https://github.com/harp-tech/protocol>. To reserve a range or certain IDs for your project or company, please follow the instructions in this repository. If the device doesn’t have a pre-allocated ID on the IDs list, this register should be set as 0.
+
+```mermaid
+---
+displayMode: compact
+---
+
+gantt
+    title R_WHO_AM_I (000) [U16]
+
+    dateFormat X
+    axisFormat %
+    tickInterval 0
+
+    section Bit
+    15-8      :byte1, 0, 1
+    7-0      :byte0, after byte1  , 2
+    section Id
+    WHO_AM_I      :id1, 0, 2
+    section Default
+    0      :d1, 0, 2
+```
+
+#### **`R_TIMESTAMP_SECOND` (U32) – System timestamp (seconds)**
+
+Address: `008`
+
+Contains the current system timestamp in whole seconds. The default value is 0 (ZERO) and will increment one unit for each elapsed second.
+
+```mermaid
+
+---
+displayMode: compact
+---
+
+gantt
+    title R_TIMESTAMP_SECONDS (008) [U32]
+
+    dateFormat X
+    axisFormat %
+    tickInterval 0
+
+    section Bit
+    31-24      :byte3, 0, 1
+    23-16      :byte2, after byte3  , 2
+    15-8      :byte1, after byte2  , 3
+    7-0      :byte0, after byte1  , 4
+    section Id
+    SECONDS      :id1, 0, 4
+    section Default
+    0      :d1, 0, 4
+```
+
+#### **`R_TIMESTAMP_SECOND` (U16) – System timestamp (seconds)**
```

Comment:
R_TIMESTAMP_SECOND (U16) – System timestamp (seconds), shouldn't be microseconds?